mitodl/micromasters
# Refactor docker-compose.yml files

#### What are the relevant tickets?
Fixes #3642 

#### What's this PR do?
Refactors docker-compose.yml to split into four files which are merged together in the order they appear on the command line. This should reduce duplication of various setup we shared between `docker-compose.yml` and `travis-docker-compose.yml`.

Also, the chrome, selenium and hub containers are removed and will only be included when running the selenium tests explicitly.

#### How should this be manually tested?
Run `docker-compose up` and `docker-compose run web bash`. You should not notice any changes except the missing selenium related containers. You should be able to make changes in docker to `/src` and have them show up locally, to confirm that the volume setup is unchanged.

Also run `./scripts/test/run_selenium_tests_dev.sh`. This should run the tests and exit without error.
